src/bindings/python/tests/test_runtime/test_model.py

```diff
@@ -748,6 +748,7 @@ def test_model_add_remove_result_parameter_sink():
 
     results = model.get_results()
     assert len(results) == 2
+    assert results.index(result2) == 1
```

Comment:
There is also `model.get_result_index(parameter)` available in Python and C++ OV API.

https://github.com/openvinotoolkit/openvino/blob/68f7d336e5a5ffaec88344b7f2eb36738b0306a4/src/bindings/python/src/pyopenvino/graph/model.cpp#L848-L852